examples/howto/data_container.myst.md

```diff
@@ -11,43 +11,103 @@ kernelspec:
 
-In the following example, this is demonstrated with fictional temperature observations.
+These containers make it extremely easy to work with data in a PyMC model. They offer a range of benefits, including:
+
+1. Visualization of data as a component of your probabilistic graph
+2. Access to labeled dimensions for readability and accessibility
+3. Support for swapping out data for out-of-sample prediction, interpolation/extrapolation, forcasting, etc.
+
+This notebook will illustrate each of these benefits in turn, and show you the best way to integrate data into your PyMC modeling workflow. 
+
++++
+
+## Types of Data Containers
+
+ PyMC offers two data containers, depending on your needs: {class}`pymc.ConstantData` and {class}`MutableData`. Both will help you visualize how data fits into your model, store the data in an `InfereceData` for reproducibility, and give access to labeled dimenions. As the names suggest, however, only `MutableData` allows you to change your data to do out-of-sample inference tasks. This ability does, however, come with a small performance cost.
+ 
+ In past versions of PyMC, the only data container was `pm.Data`. This container is still available for backwards compatability, but the current best practice is to use either `pm.MutableData` or `pm.ConstantData`, depending on your needs. 
+
++++
+
+## Constant Data
+
+The `pm.ConstantData` is a way to add fixed data to a model. It provides a speed boost in exchange for the ability to change the data. If you don't plan on doing out-of-sample prediction, `pm.ConstantData` is for you. 
+
+In the example shows some of the differences between using a data container, in this case `pm.ConstantData`, and "raw" data. This first model shows how raw data, in this case a numpy array, can be directly provided to a PyMC model.
+
+```{code-cell} ipython3
+true_mu = 3
+true_std = 5
+x = rng.normal(loc=true_mu, scale=true_std, size=100)
+with pm.Model() as no_data_model:
+    mu = pm.Normal("mu")
+    sigma = pm.Exponential("sigma", 1)
+    obs = pm.Normal("obs", mu=mu, sigma=sigma, observed=x)
+```
+
+Looking at the resulting computational graph, the `obs` node is shaded gray to indicate that it has observed data. But the data itself is not shown on the graph.  
+
+```{code-cell} ipython3
+pm.model_to_graphviz(no_data_model)
+```
+
+In this next model, we create a `pm.ConstantData` container to hold the observations, and pass this container to the `observed`:
+
+```{code-cell} ipython3
+with pm.Model() as no_data_model:
+    data = pm.ConstantData("data", x)
```

Comment:
Do you think it may be worth to show a simple GLM so we also discuss Data for predictors?

Reply:
I sort of do that in the section "Using MutableData container variables to predict on new data" -- starting on line 262, using a Logistic regression. It could perhaps be more emphasized that it's a GLM?

Reply:
I think it's good now